### PR TITLE
DOC: Update start/end parameter for BusinessHour and CustomBusinessHour

### PR DESCRIPTION
**Pull request by gandhis1**
These are all valid usages. Examples from pandas-stubs here:

https://github.com/pandas-dev/pandas-stubs/blob/main/tests/test_timefuncs.py#L601-L639

### CODE REVIEW

File: pandas/_libs/tslibs/offsets.pyx [hunk 1]
```diff
@@ -1613,9 +1613,9 @@ cdef class BusinessHour(BusinessMixin):
         Normalize start/end dates to midnight before generating date range.
     weekmask : str, Default 'Mon Tue Wed Thu Fri'
         Weekmask of valid business days, passed to ``numpy.busdaycalendar``.
-    start : str, default "09:00"
+    start : str, time, or list of str/time, default "09:00"
```
Comment (mroeschke) [inline pandas/_libs/tslibs/offsets.pyx:1616]: Could you add to the `Examples` section showing this?

Comment (gandhis1) [reply pandas/_libs/tslibs/offsets.pyx:1618]: @mroeschke Done

File: pandas/_libs/tslibs/offsets.pyx [hunk 2]
```diff
@@ -3619,16 +3626,23 @@ cdef class CustomBusinessHour(BusinessHour):
...
     --------
     >>> ts = pd.Timestamp(2022, 8, 5, 16)
     >>> ts + pd.offsets.CustomBusinessHour()
     Timestamp('2022-08-08 09:00:00')
+    >>> ts + pd.offsets.CustomBusinessHour(start="11:00")
+    Timestamp('2022-08-08 11:00:00')
+    >>> ts + pd.offsets.CustomBusinessHour(end=dt_time(19, 0))
+    Timestamp('2022-08-05 17:00:00')
+    >>> ts + pd.offsets.CustomBusinessHour(start=[dt_time(9, 0), "20:00"],
+    ...                                    end=["17:00", dt_time(22, 0)])
```
Comment (gandhis1) [inline pandas/_libs/tslibs/offsets.pyx:3644]: The docstrings examples apparently get compiled. This is failing. I tried `time`, but the error seemed to indicate it uses the same imports as the module itself - so I updated to using `dt_time`, which still didn't work. Based on the following import:

```
from cpython.datetime cimport (
    PyDate_Check,
    PyDateTime_Check,
    PyDelta_Check,
    date,
    datetime,
    import_datetime,
    time as dt_time,
    timedelta,
)
```

I would have expected `dt_time` to work. What's the right syntax here?

File: pandas/_libs/tslibs/offsets.pyx [hunk 3]
```diff
@@ -1615,16 +1615,23 @@ cdef class BusinessHour(BusinessMixin):
...
         End time of your custom business hour in 24h format.
 
     Examples
     --------
     >>> ts = pd.Timestamp(2022, 8, 5, 16)
     >>> ts + pd.offsets.BusinessHour()
     Timestamp('2022-08-08 09:00:00')
+    >>> ts + pd.offsets.BusinessHour(start="11:00")
+    Timestamp('2022-08-08 11:00:00')
+    >>> ts + pd.offsets.BusinessHour(end=dt_time(19, 0))
```
Comment (mroeschke) [inline pandas/_libs/tslibs/offsets.pyx:1630]: You'll need to import `dt_time` before the first usage.

`>>> from datetime import time as dt_time`

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @gandhis1

**Pull request merged**